src/main/java/Dao/CategoryDao.java

```diff
@@ -0,0 +1,69 @@
+package Dao;
+
+import Entities.Category;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CategoryDao extends AbstractDao<Category> {
+
+    @Override
+    public Category getById(int id) {
+        String SQL_SELECT = "Select * from category where category_id=?";
+        try (PreparedStatement preparedStatement = this.executePrepareStatement(SQL_SELECT)) {
+            preparedStatement.setInt(1, id);
+            ResultSet rs = preparedStatement.executeQuery();
+            if (rs.next()) {
+                return new Category(rs.getLong("category_id"), rs.getString("category_name"));
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public List<Category> getAll() {
+        List<Category> categories = new ArrayList<>();
+        String SQL_SELECT_ALL = "Select * from category";
+        try(PreparedStatement ps=this.executePrepareStatement(SQL_SELECT_ALL)) {
+            ResultSet rs=ps.executeQuery();
+            while(rs.next()){
+                categories.add(new Category(rs.getLong("category_id"), rs.getString("category_name")));
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return categories;
+    }
+
+    @Override
+    public void create(Category category) {
+        String SQL_INSERT = "Insert into categories (category_id, customer_name,) values(?,?)";

```

Comment:
is category and categories is two different tables or the same one in your db? If it's the same one the name does not match with the name in the methods above.

Reply:
Yeah that's same, first I created all the tables in singular worked on the project with that than at a point I had to change tables into plural since data in all tables will not be just one, thanks again for pointing on it, changed all of them